File: backend/app/api/v1/intent.py
"""意图路由（LLD §4.2.8）：``POST /intent/classify``。

独立于编排图的入口分流接口：研究路径由前端随后调用 ``POST /runs``，
闲聊路径由 ``POST /assistant/chat`` 承接；本接口不落任何项目数据。
"""

from __future__ import annotations

from fastapi import APIRouter, Request

from app.agents.intent_router import classify_intent
from app.api.deps import CurrentUser
from app.schemas.intent import IntentClassifyRequest, IntentClassifyResponse

router = APIRouter(prefix="/intent", tags=["intent"])


@router.post("/classify", response_model=IntentClassifyResponse, summary="判别用户输入意图")
async def classify(
    payload: IntentClassifyRequest,
    request: Request,
    current_user: CurrentUser,
) -> IntentClassifyResponse:
    """判别闲聊 / 研究 / 不确定；LLM 不可用时保守降级为研究。"""
    del current_user  # 仅要求登录态；M2-1 不做用户级个性化
    llm = getattr(request.app.state, "llm", None)
    decision = await classify_intent(payload.text, llm=llm, force=payload.force)
    return IntentClassifyResponse(
        intent=decision.intent,
        confidence=decision.confidence,
        recommended_template=decision.recommended_template,
        recommended_tier=decision.recommended_tier,
        estimated_token_budget=decision.estimated_token_budget,
        estimated_cost_grade=decision.estimated_cost_grade,
        source=decision.source,
        degraded=decision.degraded,
        reason=decision.reason,
    )

File: frontend/src/services/api/intent.ts
// 意图路由 API（M2-1：POST /intent/classify，backend/app/api/v1/intent.py）
import { http } from '../http/http'
import type { IntentClassifyRequest, IntentClassifyResponse } from './types'

// 判别用户输入意图：chat 闲聊 / research 研究 / uncertain 不确定
// force 提供手动强制路径；LLM 不可用时后端保守降级 research 并置 degraded=true
export function classifyIntent(
  body: IntentClassifyRequest,
  options?: { idempotencyKey?: string }
): Promise<IntentClassifyResponse> {
  return http<IntentClassifyResponse>('/intent/classify', {
    method: 'POST',
    body: JSON.stringify(body),
    idempotencyKey: options?.idempotencyKey
  })
}
